shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala

```diff
@@ -350,13 +380,25 @@ class UCX(executorId: Int, usingWakeupFeature: Boolean = true) extends AutoClose
   private[ucx] def setupEndpoint(
       endpointId: Long, workerAddress: WorkerAddress, peerRkeys: Rkeys): UcpEndpoint = {
     logDebug(s"Starting/reusing an endpoint to $workerAddress with id $endpointId")
-    // create an UCX endpoint using workerAddress
+    // create an UCX endpoint using workerAddress or socket address
     endpoints.computeIfAbsent(endpointId,
       (_: Long) => {
         logInfo(s"No endpoint found for $endpointId. Adding it.")
-        val ep = worker.newEndpoint(
-          new UcpEndpointParams()
-            .setUcpAddress(workerAddress.address))
+        val epParams =  new UcpEndpointParams().setPeerErrorHandlingMode().setErrorHandler(
+          (ucpEndpoint: UcpEndpoint, i: Int, errorString: String) => {
+            logError(s"Endpoint to $endpointId got error: $errorString")
```

Comment:
Yes onError callbacks should be called for send/rma/flush request on endpoint. But receive is per worker - so no onError, unless error happened in between receive

Reply:
Can we `setPeerErrorHandlingMode` if and only if we have `rapidsConf.shuffleUcxUseSockaddr`? Does it work without it? And are there negative effects of peer error handling mode? I thought in the past there were issues where certain transports would not be enabled with peer error handling.

Reply:
`setPeerErrorHandlingMode` and `sockAddr` are not mutually necessary. Yes, some shared memory transports may not be selected with PeerErrorHandling. But peerErrorHanlder may work even w/o sockaddress connection.

Reply:
> Yes, some shared memory transports may not be selected with PeerErrorHandling

Which ones specifically (i.e.does it include cuda_ipc?) In any case, it does seem we need another config to set this up. So we have a way to remove it in case we really want that shared memory transport.

Reply:
No cuda_ipc support peer failure: https://github.com/openucx/ucx/blob/master/src/uct/cuda/cuda_ipc/cuda_ipc_iface.c#L130

Reply:
E.g. gdr_copy doesn't have it: https://github.com/openucx/ucx/blob/master/src/uct/cuda/gdr_copy/gdr_copy_iface.c#L60

Reply:
What about `cuda_copy` does it need it too?

https://github.com/openucx/ucx/blob/master/src/uct/cuda/cuda_copy/cuda_copy_iface.c#L72

Reply:
@Akshay-Venkatesh correct me if i wrong, but cuda_copy is 1 peer interface, so there's no "remote peer". There's PR to enable cuda_copy between peers: https://github.com/openucx/ucx/pull/6115

Reply:
@petro-rudenko  openucx/ucx#6115 is not for remote peer. It's still all within the process domain.

Reply:
Correct, forgot about multigpu in 1 process. So yes, cuda_copy is within 1 process, so no remote peer failure handle needed